### DISCUSSION TITLE
Unbound Type

### DISCUSSION THREAD

**Discussion by dashne** [Washi1337/AsmResolver#552] [General]
how to check if a type is an unbound type (a generic type, but no type arguments were provided)

**Comment by Washi1337** [comment]
So an ordinary `TypeDefinition` with `GenericParameters.Count > 0`?

**Comment by dashne** [comment]
what if type argument is provided?

**Reply by Washi1337** [reply]
Then this would not have been a `TypeDefinition` instance but a `GenericInstanceTypeSignature` that is (potentially) wrapped in a `TypeSpecification` instance.
